src/calliope/preprocess/model_math.py

```diff
@@ -1,55 +1,56 @@
 """Calliope math handling with interfaces for pre-defined and user-defined files."""
 
+import importlib.resources
 import logging
 from copy import deepcopy
-from importlib.resources import files
 from pathlib import Path
 
-from calliope import exceptions
 from calliope.attrdict import AttrDict
+from calliope.exceptions import ModelError
 from calliope.util.schema import MATH_SCHEMA, validate_dict
 from calliope.util.tools import relative_path
 
 LOGGER = logging.getLogger(__name__)
-MATH_DIR = files("calliope") / "math"
 
 
-class ModelMath:
-    """Calliope math preprocessing."""
+class CalliopeMath:
+    """Calliope math handling."""
 
     ATTRS_TO_SAVE = ("history", "data")
+    ATTRS_TO_LOAD = ("history",)
 
     def __init__(
-        self,
-        math_to_add: str | list | dict | None = None,
-        model_def_path: str | Path | None = None,
+        self, math_to_add: list[str | dict], model_def_path: str | Path | None = None
     ):
         """Calliope YAML math handler.
 
-        Can be initialised in the following ways:
-        - default: 'plan' model math is loaded.
-        - list of math files: pre-defined or user-defined math files.
-        - dictionary: fully defined math dictionary with configuration saved as keys (see `ATTRS_TO_SAVE`).
-
         Args:
-            math_to_add (str | list | dict | None, optional): Calliope math to load. Defaults to None (only base math).
-            model_def_path (str | Path | None, optional): Model definition path, needed for user math. Defaults to None.
+            math_to_add (list[str | dict]):
+                List of Calliope math to load.
+                If a string, it can be a reference to pre-/user-defined math files.
+                If a dictionary, it is equivalent in structure to a YAML math file.
+            model_def_path (str | Path | None, optional): Model definition path, needed when using relative paths. Defaults to None.
         """
         self.history: list[str] = []
-        self.data: AttrDict = AttrDict()
-        if math_to_add is None:
-            math_to_add = []
-        elif isinstance(math_to_add, str):
-            math_to_add = [math_to_add]
-
-        if isinstance(math_to_add, list):
-            self._init_from_list(["plan"] + math_to_add, model_def_path)
-        else:
-            self._init_from_dict(math_to_add)
+        self.data: AttrDict = AttrDict(
+            {
+                "variables": {},
```

Comment:
Should be `TypedDict` above?
The backend also refers to through `ORDERED_COMPONENTS_T`, so maybe the order should be defined here so that we have it in one tidy place.